scripts/prep-t2m-v4.py

```
@@ -0,0 +1,307 @@
+    def qmul_arr(a, b):
+        ax, ay, az, aw = a[:, 0], a[:, 1], a[:, 2], a[:, 3]
+        bx, by, bz, bw = b[:, 0], b[:, 1], b[:, 2], b[:, 3]
+        o = np.empty_like(a)
+        o[:, 0] = aw*bx + ax*bw + ay*bz - az*by
+        o[:, 1] = aw*by - ax*bz + ay*bw + az*bx
+        o[:, 2] = aw*bz + ax*by - ay*bx + az*bw
+        o[:, 3] = aw*bw - ax*bx - ay*by - az*bz
+        return o
+    world = {}
+    def world_of(j):
+        if j in world: return world[j]
```

Ruff (0.15.20)

[error] Multiple statements on one line (colon)

(E701)